sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -0,0 +1,514 @@
+     * {@link EncryptionAlgorithm#A192CBC A192CBC}, {@link EncryptionAlgorithm#A192CBC_HS384 A192CBC-HS384}, {@link EncryptionAlgorithm#A256CBC A256CBC} and {@link EncryptionAlgorithm#A256CBC_HS512 A256CBC-HS512} </p>
+     *
+     * @param algorithm The algorithm to be used for encryption.
+     * @param plaintext The content to be encrypted.
+     * @throws ResourceNotFoundException if the key cannot be found for encryption.
+     * @return A {@link Mono} containing a {@link EncryptResult} whose {@link EncryptResult#cipherText() cipher text} contains the encrypted content.
+     */
+    public Mono<EncryptResult> encrypt(EncryptionAlgorithm algorithm, byte[] plaintext) {
+        return withContext(context -> encrypt(algorithm, plaintext, context, null, null));
+    }
+
+    /**
+     * Encrypts an arbitrary sequence of bytes using the configured key. Note that the encrypt operation only supports a
+     * single block of data, the size of which is dependent on the target key and the encryption algorithm to be used. The encrypt
+     * operation is supported for both symmetric keys and asymmetric keys. In case of asymmetric keys public portion of the key is used
+     * for encryption. This operation requires the keys/encrypt permission.
+     *
+     * <p>The {@link EncryptionAlgorithm encryption algorithm} indicates the type of algorithm to use for decrypting the specified encrypted content. Possible values
+     * for assymetric keys include: {@link EncryptionAlgorithm#RSA1_5 RSA1_5}, {@link EncryptionAlgorithm#RSA_OAEP RSA_OAEP} and {@link EncryptionAlgorithm#RSA_OAEP_256 RSA_OAEP_256}.
+     * Possible values for symmetric keys include: {@link EncryptionAlgorithm#A128CBC A128CBC}, {@link EncryptionAlgorithm#A128CBC_HS256 A128CBC-HS256},
+     * {@link EncryptionAlgorithm#A192CBC A192CBC}, {@link EncryptionAlgorithm#A192CBC_HS384 A192CBC-HS384}, {@link EncryptionAlgorithm#A256CBC A256CBC} and {@link EncryptionAlgorithm#A256CBC_HS512 A256CBC-HS512} </p>
+     *
+     * @param algorithm The algorithm to be used for encryption.
+     * @param plaintext The content to be encrypted.
+     * @param iv The initialization vector
+     * @param authenticationData The authentication data
+     * @throws ResourceNotFoundException if the key cannot be found for encryption.
+     * @return A {@link Mono} containing a {@link EncryptResult} whose {@link EncryptResult#cipherText() cipher text} contains the encrypted content.
+     */
+    public Mono<EncryptResult> encrypt(EncryptionAlgorithm algorithm, byte[] plaintext, byte[] iv, byte[] authenticationData) {
+        return withContext(context -> encrypt(algorithm, plaintext, context, iv, authenticationData));
+    }
+
+    Mono<EncryptResult> encrypt(EncryptionAlgorithm algorithm, byte[] plaintext, Context context, byte[] iv, byte[] authenticationData) {
+        Objects.requireNonNull(algorithm);
+
+        if(key == null){
+            this.key = getKey().block().value().keyMaterial();
+        }
```

Comment:
getKey might fail if the calling principal doesn't have the key_get permission.  This should be handled and default to calling the service for all crypto calls when no key is avialable.

Reply:
added the logic to handle this issue and default to service api if key not available locally.